internal/controller/application_controller.go

```diff
@@ -326,3 +352,53 @@ func (r *ApplicationPullSecretCreator) doesApplicationPullSecretExist(ctx contex
 
 	return true, nil
 }
+
+// unlinkApplicationSecretFromServiceAccount ensures that the given secret is not linked with the namespace service account.
+func (r *ApplicationPullSecretCreator) unlinkApplicationSecretFromServiceAccount(ctx context.Context, secretNameToRemove, namespace string) error {
```

Comment:
The name is quite confusing, since it says `*FromServiceAccount`, which implies generic case. However, you operate on `NamespaceServiceAccountName` SA only.

Reply:
ok